### PR TITLE
Add fuse0 to ATxmega128A4U/64A4U/32A4U/16A4U

### PR DESCRIPTION
**Pull request by stefanrueger**
See page 29 of Atmel's 8077I–AVR–11/2012 [XMEGA A MANUAL](https://ww1.microchip.com/downloads/en/DeviceDoc/doc8077.pdf)

Screenshot from 2023-05-20 01-18-41

### CODE REVIEW

**Comment by mcuee** [comment]
Yes this PR is good.

```
PS C:\work\avr\avrdude_test\avrdude_bin> .\avrdude_pr1360 -C .\avrdude_pr1360.conf -c avrispmkii -p x32a4u -t
avrdude_pr1360: AVR device initialized and ready to accept instructions
avrdude_pr1360: device signature = 0x1e9541 (probably x32a4u)
avrdude> part
AVR Part                      : ATxmega32A4U
RESET disposition             : dedicated
RETRY pulse                   : SCK
Serial program mode           : yes
Parallel program mode         : yes
Memory Detail                 :

                                  Block Poll               Page                       Polled
  Memory Type Alias    Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
  ----------- -------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
  fuse1                   0     0     0    0 no          1    1      0     0     0 0x00 0x00
  fuse2                   0     0     0    0 no          1    1      0     0     0 0x00 0x00
  fuse4                   0     0     0    0 no          1    1      0     0     0 0x00 0x00
  fuse5                   0     0     0    0 no          1    1      0     0     0 0x00 0x00
  lock                    0     0     0    0 no          1    1      0     0     0 0x00 0x00
  signature               0     0     0    0 no          3    1      0     0     0 0x00 0x00
  prodsig                 0     0     0    0 no         50   50      0     0     0 0x00 0x00
  data                    0     0     0    0 no          0    1      0     0     0 0x00 0x00
  fuse0                   0     0     0    0 no          1    1      0     0     0 0x00 0x00
  eeprom                  0     0     0    0 no       1024   32      0     0     0 0x00 0x00
  flash                   0     0     0    0 no      36864  256      0     0     0 0x00 0x00
  application             0     0     0    0 no      32768  256      0     0     0 0x00 0x00
  apptable                0     0     0    0 no       4096  256      0     0     0 0x00 0x00
  boot                    0     0     0    0 no       4096  256      0     0     0 0x00 0x00
  usersig                 0     0     0    0 no        256  256      0     0     0 0x00 0x00
avrdude> dump fuse0
Reading | ################################################## | 100% 0.00 s
0000  ff                                                |.               |
avrdude> write fuse0 0 0x55
Writing | ################################################## | 100% 0.01 s
avrdude> flush
avrdude> dump fuse0
Reading | ################################################## | 100% 0.00 s
0000  55                                                |U               |
avrdude> quit

avrdude_pr1360 done.  Thank you.

```

**Comment by stefanrueger** [comment]
Turns out `ATxmega-B` also have `fuse0` even though `FUSEBYTE0` is not mentioned in the XMEGA B Manual. `JTAG user ID` is alluded to in individual ATxmega B part datasheets, though:

Screenshot from 2023-05-20 12-20-48
 
And `avrdude.conf` gets this right:
 - All ATxmega `fuse0` memories are for A/B parts:
   ```
   $ avrdude -patxmega*/At | grep ^.ptmm | cut -f2-3  | grep fuse0 | sort -u | grep -v "^ATxmega[0-9]*[AB]"
   ```
 - All  ATxmega A/B parts have a `fuse0` memory
   ```
   $ avrdude -patxmega*/At | grep "^.ptmm.*ATxmega[0-9]*[AB]" | grep fuse0 | cut -f2 | sort -u | wc -l
   29
   $ avrdude -patxmega*/At | grep "^.ptmm.*ATxmega[0-9]*[AB]" | cut -f2 | sort -u | wc -l
   29
   ```

**Comment by mcuee** [comment]
I think this PR is good to go.

File: src/avrdude.conf.in [hunk 1]
```diff
@@ -15769,6 +15786,8 @@ part parent "x16a4u"
     mcuid                  = 233;
     signature              = 0x1e 0x94 0x43;
 
+    memory "fuse0" = NULL;
```
Comment (MCUdude) [inline src/avrdude.conf.in:15789]: Shouldn't `= NULL` be aligned with the other values?

**Pull request merged**